pkg/storage/allocator_scorer.go

```diff
@@ -1208,11 +1224,11 @@ func underfullRangeThreshold(options scorerOptions, mean float64) float64 {
 }
 
 func overfullThreshold(mean float64, thresholdFraction float64) float64 {
-	return mean * (1 + thresholdFraction)
+	return mean + math.Max(mean*thresholdFraction, minRangeRebalanceThreshold)
 }
 
 func underfullThreshold(mean float64, thresholdFraction float64) float64 {
-	return mean * (1 - thresholdFraction)
+	return mean - math.Max(mean*thresholdFraction, minRangeRebalanceThreshold)
```

Comment:
Does anything interesting happen here if we end up <0? Say the mean is 1, we end up with -1. I guess an empty node may never receive a replica? Which is honestly fine in that example.

Reply:
No, nothing happens if it is negative. This is used to compare with the actual number of replicas and take an action if the actual number is below the threshold. When negative - no action is taken so we won't try to rebalance (due to the specific reason of being away from the mean).